src/Init/Tactics.lean

```diff
@@ -461,11 +461,14 @@ syntax config := atomic(" (" &"config") " := " withoutPosition(term) ")"
 /-- The `*` location refers to all hypotheses and the goal. -/
 syntax locationWildcard := " *"
 
+/-- The `⊢` location refers to the current goal. -/
+syntax locationTurnstile := patternIgnore(atomic("|" noWs "-") <|> "⊢")
```

Comment:
What do you think about calling this `locationType` to mirror the field in `Location.target`? The fact that the symbol is known as a turnstile is a little bit of trivia.

I'm not sure what we've decided for naming the parts of a goal. I've seen "goal", "target", and "goal type" for the type of a goal. Using "goal" for the goal's type might be acceptable synecdoche.

We don't have to solve the issue for the docstrings in this PR. But, as you know from bootstrapping issues, it would be nice to at least not have to rename the `syntax` definition later!

Reply:
Thanks, I've made this change!